src/analyses/Makefile

```diff
@@ -1,5 +1,6 @@
 SRC = ai.cpp \
       ai_domain.cpp \
+      ai_history.cpp \
```

Comment:
Commit message on 7ba3feb41979dd9b6763cc2825a559e75641e1fb: s/compatable/compatible

Also, this seems like it could be a whole separate PR on its own

Reply:
It's really the key commit for this whole PR.  All of the other commits are just there to make it easier / simpler.  I didn't want to submit those separately as many of them are kind of pointless on their own.  There have already been maybe 4 PRs broken out of this, with more to come.